python/cudf/cudf/core/frame.py

```diff
@@ -2280,19 +2312,42 @@ def drop_duplicates(
         if len(subset_cols) == 0:
             return self.copy(deep=True)
 
-        result = self.__class__._from_data(
-            *libcudf.stream_compaction.drop_duplicates(
-                self,
-                keys=subset,
-                keep=keep,
-                nulls_are_equal=nulls_are_equal,
-                ignore_index=ignore_index,
-            )
+        keys = self._positions_from_column_names(
+            subset, include_index=not ignore_index
+        )
+        result = self._drop_duplicates(
+            keys=keys,
+            keep=keep,
+            nulls_are_equal=nulls_are_equal,
+            ignore_index=ignore_index,
         )
-
         result._copy_type_metadata(self)
         return result
 
+    def _positions_from_column_names(self, column_names, include_index=False):
```

Comment:
Ok this is awesome! I was assuming mypy is unhappy about this but I'm very glad we can remove this argument.

Reply:
Synced up offline, the challenge here is that there are methods defined in `frame.py` that needs to be "index aware". Like `drop_duplicates` takes `ignore_index` as input. It makes it pretty awkward that helper functions in `frame.py` also needs to be "index aware". The way to go is to move these functions to `indexed_frames` and have specific implementation for them in `Index`.

Reply:
mypy basically thinks it's a bad idea (and I agree) to allow cases where a child class cannot be interchangeably used with a parent.
```
class Parent:
    def foo(x: Sequence):
        ...
class Child(Parent):
    def foo(x: list):  <-- list is a subset of sequence
        ...
def bar(p: Parent, x: Sequence):
    p.foo(x)  <-- This won't work if p is actually a Child and x is a non-list Sequence
```
This example is for the case where the child accepts a narrower type than the parent, but the same goes for a child class not being able to handle all the parameters of a parent class and various other cases. This property is known as the [Liskov substitution principle](https://en.wikipedia.org/wiki/Liskov_substitution_principle), and to a large extent it's already basically enforced by most compiled languages including C++ AFAIK. Python muddies the waters because it's so flexible and you often want to ducktype, but it's very easy to ducktype in type-unsafe ways for reasons like this.